### DISCUSSION TITLE
How to upload selected file (to Google Firestore) using QUploader?

### DISCUSSION THREAD

**Discussion by eugenevk** [quasarframework/quasar#12347] [General - Components / Directives / etc]
I am using Firebase Storage from Google to store my project's images. I need to access Firebase Storage with the right credentials, so I have created the following program for this:
```
import { fbStorage } from 'boot/firebase'
import { getDownloadURL, ref, uploadBytes } from 'firebase/storage'

let error = null
let url = null
let path = null

const uploadFile = async (file, companyId) => {
  path = `companies/${companyId}/${file.name}`

  const storageRef = ref(fbStorage, path)

  try {
    await uploadBytes(storageRef, file).then(snapshot => {
      url = getDownloadURL(storageRef).then(snapshot => {
        console.log('Uploaded a file! URL=', url)
      })
    })
  } catch (err) {
    error = err.message
    console.error('Error storing file in firestore', error)
  }

  return {
    url,
    path
  }
}

export default uploadFile
```
I am using QUploader to let the user select the file. Now I need to pass the file object to this function.
Two questions:
1. How can I get the file object from QUploader?
2. How can I avoid the user has to click the "Upload" button from QUploader to upload the file, but use the save function in my component instead?

**Comment by jrhopkins83** [comment]
Not sure if this is what you're looking for but this article uses a custom uploader: FBQUploader.  See section 5.2.
https://dev.to/quasar/to-the-stars-with-quasar-firebase-user-profile-319l#cloudStorage

**Comment by BenJackGill** [comment]
@eugenevk Did you ever figure this out? I am having the same issue. The article linked by @jrhopkins83 doesn't seem to help me either.

**Reply by eugenevk** [reply]
Hi @BenJackGill , no I haven't progressed on it. Put it aside for the time being, since I had/have other things to do that are more important.
